src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -0,0 +1,3138 @@
+void AutomationInstrumentClipView::exitScaleMode() {
+	int32_t scrollAdjust = instrumentClipView.setupForExitingScaleMode();
+
+	char modelStackMemory[MODEL_STACK_MAX_SIZE];
+	ModelStackWithTimelineCounter* modelStack = currentSong->setupModelStackWithCurrentClip(modelStackMemory);
+	InstrumentClip* clip = (InstrumentClip*)modelStack->getTimelineCounter();
+
+	// See which NoteRows need to animate
+	PadLEDs::numAnimatedRows = 0;
+	for (int32_t i = 0; i < clip->noteRows.getNumElements(); i++) {
+		NoteRow* thisNoteRow = clip->noteRows.getElement(i);
+		int32_t yDisplayTo = thisNoteRow->y - (clip->yScroll + scrollAdjust);
+		clip->inScaleMode = true;
+		int32_t yDisplayFrom = clip->getYVisualFromYNote(thisNoteRow->y, currentSong) - clip->yScroll;
+		clip->inScaleMode = false;
+
+		// If this NoteRow is going to end up on-screen or come from on-screen...
+		if ((yDisplayTo >= 0 && yDisplayTo < kDisplayHeight) || (yDisplayFrom >= 0 && yDisplayFrom < kDisplayHeight)) {
+			uint8_t mainColour[3];
+			uint8_t tailColour[3];
+			uint8_t blurColour[3];
+			clip->getMainColourFromY(thisNoteRow->y, thisNoteRow->getColourOffset(clip), mainColour);
+			getTailColour(tailColour, mainColour);
+			getBlurColour(blurColour, mainColour);
+
+			ModelStackWithNoteRow* modelStackWithNoteRow = modelStack->addNoteRow(thisNoteRow->y, thisNoteRow);
+
+			instrumentClipView.drawMuteSquare(thisNoteRow, PadLEDs::imageStore[PadLEDs::numAnimatedRows],
+			                                  PadLEDs::occupancyMaskStore[PadLEDs::numAnimatedRows]);
+
+			PadLEDs::numAnimatedRows++;
+			if (PadLEDs::numAnimatedRows >= kMaxNumAnimatedRows) {
+				break;
+			}
+		}
+	}
+
+	clip->yScroll += scrollAdjust;
+
+	instrumentClipView.recalculateColours();
+	setLedStates();
+}
+
+//pad action
+//handles shortcut pad action for automation (e.g. when you press shift + pad on the grid)
+//everything else is pretty much the same as instrument clip view
+
+ActionResult AutomationInstrumentClipView::padAction(int32_t x, int32_t y, int32_t velocity) {
+
+	InstrumentClip* clip = getCurrentClip();
+	Instrument* instrument = (Instrument*)clip->output;
+
+	char modelStackMemory[MODEL_STACK_MAX_SIZE];
+	ModelStackWithTimelineCounter* modelStack = currentSong->setupModelStackWithCurrentClip(modelStackMemory);
+
+	// Edit pad action...
+	if (x < kDisplayWidth) {
+		if (sdRoutineLock) {
+			return ActionResult::REMIND_ME_OUTSIDE_CARD_ROUTINE;
+		}
+
+		//if the user wants to change the parameter they are editing using Shift + Pad shortcut
+		if (velocity) {
+
+			if (Buttons::isShiftButtonPressed()) {
+
+				handleSinglePadPress(modelStack, clip, x, y, true);
+
+				return ActionResult::DEALT_WITH;
+			}
+			//regular automation step editing action
+			else {
+				goto doRegularEditPadActionProbably;
+			}
+		}
+
+		// Regular edit-pad action
+		else {
+doRegularEditPadActionProbably:
+			if (isUIModeWithinRange(editPadActionUIModes)) {
```

Comment:
Can remove the `else` cases and just let control fall through the ifs, since there is an early return in the only case where we don't do this check + `editPadAction`. That allows removing the `goto`.